Resources/public/js/recommendationrestclient.js

```diff
@@ -59,12 +60,29 @@
             }
         };
 
-        var attributes = '';
         for (var i = 0; i < this.attributes.length; i++) {
             attributes = attributes + '&attribute=' + this.attributes[i];
         }
 
-        xmlhttp.open('GET', this.endpointUrl + this.scenario + '.json' + '?numrecs=' + this.limit + '&' + 'contextitems=' + this.contextItems + attributes + '&contenttype=' + this.contentType + '&outputtype=' + this.outputType + '&categorypath=' + encodeURIComponent(this.categoryPath) + '&lang=' + this.language, true);
+        requestQueryString.push(
```

Comment:
I would do it this way: `requestQueryString = [ ... values ...]; ` instead of pushing these values to the array.
In this case, the `requestQueryString` variable declaration at the top of function body will be undefined (will not have an empty array by default).